pkg/kubernetes/pod/v1alpha1/pod.go

```diff
@@ -142,3 +142,35 @@ func IsNil() Predicate {
 func (p *Pod) GetAPIObject() *corev1.Pod {
 	return p.object
 }
+
+// FromList created a PodList with provided api podlist
+func FromList(pods *corev1.PodList) *PodList {
+	pl := ListBuilderForAPIList(pods).List()
+	return pl
+}
+
+// GetNodeNames returns the nodes on which pods are running
+func (pl *PodList) GetNodeNames() []string {
+	nodeNames := []string{}
+	for _, p := range pl.items {
+		nodeNames = append(
+			nodeNames,
+			p.object.Spec.NodeName,
+		)
+	}
+	return nodeNames
+}
+
+// IsMatchesAnyNode checks the PodList is running on the provided nodes
+func (pl *PodList) IsMatchesAnyNode(nodes []string) bool {
+	var result bool
+	for _, p := range pl.items {
+		result = false
```

Comment:
why is this needed?

Reply:
Here it is checking for each pod that its node is present in the previous nodeNames or not. So for each pod it starts with false and if found it turns it to true.